pkg/network/common/egressip.go

```diff
@@ -242,6 +291,24 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *osdnv1.HostSubnet) {
 		delete(eit.nodesByNodeIP, node.nodeIP)
 	}
 
+	// We need to handle the case where the SDN pods restart, upon which both
+	// the Node will have the annotation set and all resources will already be
+	// existing, meaning: we might receive the HostSubnet event after the Node
+	// event, which means we need to lookup the Node annotation as to sync our
+	// data correctly. On normal start-up this will not be the case since the
+	// annotation will be set after any HostSubnet event is received.
+	if eit.nodeInformer != nil && node.capacity == unlimitedNodeCapacity {
+		v1Node, err := eit.nodeInformer.Lister().Get(hs.Name)
+		if err != nil {
+			klog.Errorf("Unable to list HostSubnet %q as to set its IP capacity, err: %v", hs.Name, err)
+			return
+		}
+		if err := eit.initNodeCapacity(v1Node); err != nil {
```

Comment:
`UpdateHostSubnetEgress` will still need to look up the `corev1.Node` since we'll need to parse the annotation on the node object in `getNodeCloudEgressIPConfig`. But I what you're saying does reduce a call to `getNodeInternalIP` from `UpdateHostSubnetEgress` so I will implement that.